src/Http/RouteCollection.php

```diff
@@ -63,19 +68,48 @@ public function delete($path, $name, $handler)
 
     public function addRoute($method, $path, $name, $handler)
```

Comment:
If routes have already been applied, should the new one be applied too? A few options of doing this:

- Refactor `$this->applyRoutes()` to process routes individually. I don't really like that
- Treat $this->routes as `$this->pendingRoutes`, and clear it at the end of `$this->applyRoutes()`. Then we can just call `$this->applyRoutes()` again.

Reply:
Second option sounds good